plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -2912,6 +2949,80 @@ public void finishStatisticsCollection(ConnectorSession session, ConnectorTableH
                 Optional.of(physicalColumnNameMapping));
     }
 
+    private void generateMissingFileStatistics(ConnectorSession session, DeltaLakeTableHandle tableHandle, Collection<ComputedStatistics> computedStatistics)
+    {
+        Map<String, AddFileEntry> addFileEntriesWithNoStats = transactionLogAccess.getActiveFiles(
+                        getSnapshot(tableHandle.getSchemaTableName(), tableHandle.getLocation(), session), session)
+                .stream()
+                .filter(addFileEntry -> addFileEntry.getStats().isEmpty()
+                        || addFileEntry.getStats().get().getNumRecords().isEmpty()
+                        || addFileEntry.getStats().get().getMaxValues().isEmpty()
+                        || addFileEntry.getStats().get().getMinValues().isEmpty()
+                        || addFileEntry.getStats().get().getNullCount().isEmpty())
+                .filter(addFileEntry -> !URI.create(addFileEntry.getPath()).isAbsolute()) //Support only relative path in AddFileEntry
```

Comment:
That would fail `ANALYZE` in case we have missing statistics in table with any absolute paths, do we want that?
I would rather skip such file from processing, and still generate extended statistics.

Reply:
We prefer exceptions to suppressing in this project.

Reply:
Rising of exception here would break legacy `ANALZYE` functionality.